taglib-extras: make taglib-extras work with taglib > 1.9

exi opened the pull request:
the current taglib_extras code uses STRLESS to compare versions, this fails because the current version 1.10 is STRLESS then the minimum version 1.6.
I replace the comparison with VERSION_LESS which works as intended.

exi committed:
make taglib-extras work with taglib > 1.9

peti commented:
Please report this bug upstream and encourage them to release a new version that fixes this issue. I believe it's fine to merge this patch here, but there should be a comment in the Nix code that points to an upstream Gibhub issue or some other ticket to make it easier for us to track whether that patch can be removed or not.

exi committed:
amarok: fix taglib search

exi committed:
cantata: fix taglib search

exi committed:
Add comments identifying workarounds

exi commented:
@peti I fixed the same error in multiple locations and added comments indicating upstream status.

Interesting that this was not catched by any build system?

exi commented:
@peti I think this PR fails the tests for unrelated reasons. Is there any way to restart the tests?
